app/views/unattended/provisioning_templates/snippet/container_certs_setup.erb

```
@@ -0,0 +1,44 @@
+<%#
+kind: snippet
+name: container_certs_setup
+model: ProvisioningTemplate
+snippet: true
+description: |
+  This snippet sets up SSL certificates for secure communication with the Foreman registry.
+-%>
+container_certs_setup() {
+  echo "#"
+  echo "Running certificate setup for container registry"
+  echo "#"
+  hostname="<%= @registration_host %>"
+  <% raise "Unable to determine registration hostname" if @registration_host.blank? %>
+  # Create directory for certificates
+  cert_dir="/etc/containers/certs.d/$hostname"
+  echo "Creating $cert_dir directory"
+  mkdir -p "$cert_dir" || { echo "Error: Failed to create directory $cert_dir"; exit 1; }
+
+  # Link CA certificate
+  ca_cert="/etc/pki/tls/certs/ca-bundle.crt"
+  if [ -f "$ca_cert" ]; then
+    ln -sf "$ca_cert" "$cert_dir/ca-bundle.crt"
+  else
+    echo "Error: CA certificate not found at $ca_cert"
+    exit 1
```

Is it wise to stop the whole registration if this fails?

I'd expect that the whole registration either does nothing (and errors out) or runs through. So perhaps first do all the checks and then the real actions?

---

Side note: does anything special need to be done to honor the "ignore errors" option on registration?

---

The ignore_subman_errors will quietly proceed in the template if subman register fails and the container cert auth setup will fail if it doesn't find the entitlement cert and key in the subman directory.